syntax = "proto3";

package user.v1alpha;

message User {
  string id = 1;
  string firstName = 2;
  string lastName = 3;
  string email = 4;
}

service UserCRUDService {
  rpc Get (GetRequest) returns (GetResponse);
  rpc Add (AddRequest) returns (AddResponse);
  rpc Update (UpdateRequest) returns (UpdateResponse);
  rpc Delete (DeleteRequest) returns (DeleteResponse);
}

message GetRequest {
  string id = 1;
  string firstName = 2;
  string lastName = 3;
  string email = 4;
}

message GetResponse {
  repeated User user = 1;
}

message AddRequest {
  string firstName = 1;
  string lastName = 2;
  string email = 3;
}

message AddResponse {
  User user = 1;
}

message UpdateRequest {
  User user = 1;
}

message UpdateResponse {
  User user = 1;
}

message DeleteResponse {
  User user = 1;
}

message DeleteRequest {
  string id = 1;
}